src/gwt/src/org/rstudio/studio/client/application/ui/impl/DesktopApplicationHeader.java

```diff
@@ -549,15 +549,22 @@ public void execute()
          // Only provide the option to ignore the update if it's not urgent.
          if (result.getUpdateUrgency() == 0)
          {
+            // We need to use a final variable here because we're using it in an
+            // anonymous inner class
+            final boolean finalIgnoredUpdate = ignoredUpdate;
             buttonLabels.add(constants_.ignoreUpdateButtonLabel());
             elementIds.add(ElementIds.DIALOG_CANCEL_BUTTON);
             buttonOperations.add(new Operation() {
                @Override
                public void execute()
                {
-                  ignoredUpdatesState_.addIgnoredUpdate(result.getUpdateVersion());
-                  // Trigger an update to the persistent updates state file
-                  eventBus_.fireEvent(new PushClientStateEvent(true));
+                  // only run the following code if we didn't already ignore the update
+                  if (!finalIgnoredUpdate)
```

Comment:
Is there a case where an update is available and we don't want the ignore button to do anything? It looks like updating the ignored versions might be good to do.

Reply:
I think this makes sense now. Just had to think about how `ignoredUpdate` is set. The ignore action doesn't have to do anything if the version has already been ignored.